enigma/src/main/java/cuchaz/enigma/classhandle/ClassHandleProvider.java

```diff
@@ -171,7 +171,7 @@ public void destroy() {
 		private final ClassEntry entry;
 		private ClassEntry deobfRef;
 		private final List<ClassHandleImpl> handles = new ArrayList<>();
-		private Result<Source, ClassHandleError> uncommentedSource;
+		private Result<Source, ClassHandleError> commentedSource;
```

Comment:
Why is this "commentedSource" now? Looks like what this is doing right now is do initial decompilation and remapping in the same stage, and then separately re-insert javadoc afterwards if it gets changed. Why? Does CFR decompilation need to insert Javadoc while decompiling? Also, does applying javadoc twice work correctly and remove the old javadoc that was there before if it gets invalidated?

Also then I'd call this something else because otherwise it kind of implies that the "source" field is not commented. Maybe "rawSource"?

(There's also still occurrences of "uncommentedSource" that you should rename along with this field)

Reply:
Yes, for cfr insertion of javadoc is part of decompilation. Previously supplying javadoc and decompiling are separate; however, cfr has no mutable tree api like procyon so I decided to merge the 2 steps. Even for procyon merging this 2 is less costly

Reply:
Reverted these changes for now.